data-prepper-plugins/http-source-common/src/test/java/org/opensearch/dataprepper/http/codec/MultiLineJsonCodecTest.java

```diff
@@ -0,0 +1,86 @@
+package org.opensearch.dataprepper.http.codec;
+
+import com.linecorp.armeria.common.HttpData;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class MultiLineJsonCodecTest {
+    private final HttpData serializedRequest = HttpData.ofUtf8("" +
+            "{ \"index\": { \"_index\": \"test-index\", \"_id\": \"id1\" } }\n" +
+            "{ \"text\": \"text1\", \"year\": \"2013\" }");
+
+    private final HttpData serializedRequestMultipleRows = HttpData.ofUtf8("" +
+            "{ \"index\": { \"_index\": \"test-index\", \"_id\": \"id1\" } }\n" +
+            "{ \"text\": \"text1\", \"year\": \"2013\" }\n" +
+            "{ \"index\": { \"_index\": \"test-index\", \"_id\": \"id1\" } }\n" +
+            "{ \"text\": \"text1\", \"year\": \"2013\" }\n" +
+            "{ \"index\": { \"_index\": \"test-index\", \"_id\": \"id1\" } }\n" +
+            "{ \"text\": \"text1\", \"year\": \"2013\" }\n" +
+            "{ \"index\": { \"_index\": \"test-index\", \"_id\": \"id1\" } }\n" +
+            "{ \"text\": \"text1\", \"year\": \"2013\" }\n" +
+            "{ \"index\": { \"_index\": \"test-index\", \"_id\": \"id1\" } }\n" +
+            "{ \"text\": \"text1\", \"year\": \"2013\" }\n");
+
+    private final HttpData serializedRequestBad = HttpData.ofUtf8("{}\n\n{}");
+    private final HttpData serializedRequestBadWithBlanks = HttpData.ofUtf8("{}\n   \n    ");
+    private final HttpData serializedRequestBadWithWhiteSpaces = HttpData.ofUtf8("\t\n\r\f {}");
+    private final HttpData serializedRequestBadEmpty = HttpData.ofUtf8("");
+    private final HttpData serializedRequestBadEmptyNewLines = HttpData.ofUtf8("\n\n\n\n\n\n\n     \n");
+    private final HttpData serializedRequestBadInvalidJson = HttpData.ofUtf8("{\"text\":");
+
+    private final MultiLineJsonCodec multiLineJsonCodec = new MultiLineJsonCodec();
```

Comment:
Let's use the `createObjectUnderTest()` method pattern. It allows more flexibility as tests grow.

Reply:
Thanks @dlvenable. I will resolve this.